src/torchmetrics/wrappers/feature_share.py

```diff
@@ -0,0 +1,125 @@
+            cached and shared across all metrics per batch.
+
+    Example::
+        >>> import torch
+        >>> _ = torch.manual_seed(42)
+        >>> from torchmetrics.wrappers import FeatureShare
+        >>> from torchmetrics.image import FrechetInceptionDistance, KernelInceptionDistance
+        >>> # initialize the metrics
+        >>> fs = FeatureShare([FrechetInceptionDistance(), KernelInceptionDistance(subset_size=10, subsets=2)])
+        >>> # update metric
+        >>> fs.update(torch.randint(255, (50, 3, 64, 64), dtype=torch.uint8), real=True)
+        >>> fs.update(torch.randint(255, (50, 3, 64, 64), dtype=torch.uint8), real=False)
+        >>> # compute metric
+        >>> fs.compute()
+        {'FrechetInceptionDistance': tensor(15.1700), 'KernelInceptionDistance': (tensor(-0.0012), tensor(0.0014))}
+
+    """
+
+    def __init__(
+        self,
+        metrics: Union[Metric, Sequence[Metric], Dict[str, Metric]],
+        max_cache_size: Optional[int] = None,
+    ) -> None:
+        # disable compute groups because the feature sharing is more custom
+        super().__init__(metrics=metrics, compute_groups=False)
+
+        if max_cache_size is None:
+            max_cache_size = len(self)
+        if not isinstance(max_cache_size, int):
+            raise TypeError(f"max_cache_size should be an integer, but got {max_cache_size}")
+
+        try:
+            first_net = next(iter(self.values()))
+            network_to_share = getattr(first_net, first_net.feature_network)
+        except AttributeError as e:
+            raise AttributeError(
+                "Tried to extract the network to share from the first metric, but it did not have a `feature_network`"
+                " attribute. Please make sure that the metric has a attribute with that name, else it cannot be shared."
+            ) from e
+        cached_net = NetworkCache(network_to_share, max_size=max_cache_size)
+
+        # set the cached network to all metrics
+        for metric_name, metric in self.items():
+            if not hasattr(metric, "feature_network"):
+                raise AttributeError(
+                    "Tried to set the cached network to all metrics, but one of the metrics did not have a "
+                    "`feature_network` attribute. Please make sure that all metrics have a attribute with that name, "
+                    f"else it cannot be shared. Failed on metric {metric_name}."
+                )
+
+            # check if its the same network as the first metric
+            if str(getattr(metric, metric.feature_network)) != str(network_to_share):
+                rank_zero_warn(
+                    f"The network to share between the metrics is not the same for all metrics. "
+                    f"Metric {metric_name} has a different network than the first metric."
+                    "This may lead to unexpected behavior.",
```

Comment:
```suggestion
                    f"The network to share between the metrics is not the same for all metrics."
                    f" Metric {metric_name} has a different network than the first metric."
                    " This may lead to unexpected behavior.",
```